server/memphis_handlers_consumers.go

```diff
@@ -102,36 +103,42 @@ func (s *Server) createConsumerDirectCommon(c *client, consumerName, cStationNam
 	err = validateConsumerType(consumerType)
 	if err != nil {
 		serv.Warnf("[tenant: %v][user: %v]createConsumerDirectCommon at validateConsumerType: Failed creating consumer %v at station %v : %v", tenantName, userName, consumerName, cStationName, err.Error())
-		return err
+		return []int{}, err
 	}
 
 	stationName, err := StationNameFromStr(cStationName)
 	if err != nil {
 		serv.Warnf("[tenant: %v][user: %v]createConsumerDirectCommon at StationNameFromStr: Consumer %v at station %v : %v", tenantName, userName, consumerName, cStationName, err.Error())
-		return err
+		return []int{}, err
 	}
 
 	exist, user, err := memphis_cache.GetUser(userName, tenantName, false)
 	if err != nil {
 		serv.Errorf("[tenant: %v][user: %v]createConsumerDirectCommon at GetUser from cache: Consumer %v at station %v : %v", tenantName, userName, consumerName, cStationName, err.Error())
-		return err
+		return []int{}, err
 	} else if !exist {
-		serv.Warnf("[tenant: %v][user: %v] createConsumerDirectCommon at GetUser from cache: user does not exist", tenantName, userName)
-		return fmt.Errorf("user does not exist in db")
+		err := errors.New("user does not exist")
+		serv.Warnf("[tenant: %v][user: %v] createConsumerDirectCommon at GetUser from cache: %s", tenantName, userName, err.Error())
+		return []int{}, err
 	}
 
 	exist, station, err := db.GetStationByName(stationName.Ext(), user.TenantName)
 	if err != nil {
 		serv.Errorf("[tenant: %v]createConsumerDirectCommon at GetStationByName: Consumer %v at station %v : %v", tenantName, consumerName, cStationName, err.Error())
-		return err
+		return []int{}, err
 	}
 
 	if !exist {
+		if requestVersion < 2 {
+			err := errors.New("station does not exist, upgrade sdk to create station")
```

Comment:
This station does not exist, a default station can not be created automatically, please upgrade your SDK version